### PR TITLE
Bug fix/Hbase where cluse working

### PR DESCRIPTION
**Pull request by GokulBothe99**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
	- Enhanced data type handling in filters for improved flexibility.
	- Introduced methods for parsing SQL queries, including extraction of WHERE clauses and order by details.
	- Added a new `RowData` class for encapsulating row data and a comparer for sorting.

- **Bug Fixes**
	- Improved error handling during SQL execution and connection management.
	- Expanded logic for filtering field values, including trimming and type parsing.
	- Updated handling of empty byte arrays for more accurate data type inference.
	- Enhanced SQL query processing and error management.

### CODE REVIEW

File: Ginger/GingerCoreNET/Database/NoSqlBase/GingerHbase.cs [hunk 1]
```diff
@@ -18,7 +18,10 @@ limitations under the License.
 
 using Amdocs.Ginger.Common;
 using Applitools.Utils;
+using Cassandra.DataStax.Graph.Internal;
 using GingerCore.Actions;
+using Microsoft.Azure.Cosmos.Core;
+using Microsoft.CodeAnalysis.VisualBasic.Syntax;
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/GingerCoreNET/Database/NoSqlBase/GingerHbase.cs:24]: done

File: Ginger/GingerCoreNET/Database/NoSqlBase/GingerHbase.cs [hunk 2]
```diff
@@ -200,7 +205,23 @@
...
+                else if (fieldValue.Contains('.'))
+                {
+                    return new SingleColumnValueFilter(Encoding.UTF8.GetBytes(family), Encoding.UTF8.GetBytes(fieldName), compareOp, System.BitConverter.GetBytes(double.Parse(fieldValue)), filterIfMissing: true);
+                }
+                else
+                {
+                    var longBytes = System.BitConverter.GetBytes(long.Parse(fieldValue));
+                    Array.Reverse(longBytes);
+                    return new SingleColumnValueFilter(Encoding.UTF8.GetBytes(family), Encoding.UTF8.GetBytes(fieldName), compareOp, longBytes, filterIfMissing: true);
+                }
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/GingerCoreNET/Database/NoSqlBase/GingerHbase.cs:224]: not required

File: Ginger/GingerCoreNET/Database/NoSqlBase/GingerHbase.cs [hunk 3]
```diff
@@ -577,6 +595,10 @@ public static string DisplayInferredTypeAndValue(byte[] byteArray)
             {
                 return "";
             }
+            if (byteArray.All(y => y == 0))
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/Database/NoSqlBase/GingerHbase.cs:598]: Add condition to check if bytearray length is less than 9, then only check for all 0

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]
Add unit test cases to cover the functionality

File: Ginger/GingerCoreNET/Database/NoSqlBase/GingerHbase.cs (hunk 3 above)
Comment (GokulBothe99) [reply Ginger/GingerCoreNET/Database/NoSqlBase/GingerHbase.cs]: done

File: Ginger/GingerCoreNET/Database/NoSqlBase/GingerHbase.cs [hunk 4]
```diff
@@ -426,120 +454,122 @@
...
                                     }
-                                    catch (Exception)
-                                    {
-
-                                        // throw;
-                                    }
-                                    path1++;
+                                }
+                                catch (Exception)
+                                {
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/GingerCoreNET/Database/NoSqlBase/GingerHbase.cs:521]: handled

File: Ginger/GingerCoreNET/Database/NoSqlBase/GingerHbase.cs [hunk 5]
```diff
@@ -426,120 +454,122 @@
...
+                                familyName = i.name;
+                                scanner = getScanner(wherepart, familyName);
+                                scanInfo = actionClient.CreateScannerAsync(table, scanner, requestOption).Result;
+
+                                if ((next = actionClient.ScannerGetNextAsync(scanInfo, requestOption).Result) != null)
+                                {
+                                    break;
+                                }
+
+                            }
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/GingerCoreNET/Database/NoSqlBase/GingerHbase.cs:476]: handled

File: Ginger/GingerCoreNET/Database/NoSqlBase/GingerHbase.cs [hunk 6]
```diff
@@ -18,17 +18,26 @@ limitations under the License.
 
 using Amdocs.Ginger.Common;
 using Applitools.Utils;
+using Cassandra.DataStax.Graph.Internal;
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/Database/NoSqlBase/GingerHbase.cs:21]: Remove unwanted usings

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**